SetK.md

```diff
@@ -0,0 +1,149 @@
+autoscale: true
+footer:  [@calvellido](https://twitter.com/calvellido) [@47deg](https://twitter.com/47deg) :: [Λrrow](http://arrow-kt.io) :: [http://arrow-kt.io/docs/datatypes/setK/](http://arrow-kt.io/docs/datatypes/setK/)
+slidenumbers: true
+
+# SetK
+
+__`SetK`__ is a data type used in __Λrrow__ as a higher kinded wrapper around the Kotlin Set collection interface.
+
+---
+
+# SetK :: Constructor
+
+__`SetK`__ can be created by directly passing a __Set__.
+
+```kotlin
+SetK(setOf(1, 2, 5, 3, 2))
+// SetK(set=[1, 2, 5, 3])
+```
+
+---
+
+# SetK :: k()
+
+It can also be created invoking the __`k()`__ function on a __Set__.
+
+```kotlin
+setOf(1, 2, 5, 3, 2).k()
+// SetK(set=[1, 2, 5, 3])
+```
+
+---
+
+# SetK :: pure()
+
+Or through the __`pure()`__ function on __SetK__.
+
+```kotlin
+SetK.pure(1)
+// SetK(set=[1])
+```
+
+---
+
+# SetK :: Instances
+
+__`SetK`__ auto derives the following different type classes instances, giving it all of their capabilities.
+
+* `Semigroup`
+* `SemigroupK`
+* `Monoid`
+* `MonoidK`
+* `Foldable`
+
+---
+
+# SetK :: Semigroup
+
+A __`Semigroup`__ for some given type `A` has a single operation (called __`combine`__), which takes two values of type `A`, and returns a value of type `A`. This operation must be guaranteed to be associative.
+
+The same can be said about `SemigroupK`, but for data types instead of objects.
```

Comment:
The main difference between `Semigroup` and `SemigroupK` is that while `Semigroup` operates on the values contained in the type contructor `SemigroupK` operates exclusively over the entire structure and not the individual items. I believe `SemigroupK` and `MonoidK` for set deserve their own slides where it's shown how `combineK and `empty` behave on a `SetK`